# [LAB-07] 3. 단일표본 T-Test - 연습문제

당신은 매일 아침 동네 빵집에서 식빵을 구입해 왔다.

이 식빵의 무게는 `500g`이라고 표시되어 있다. 하지만 당신은 자신이 구매하는 빵이 `500g`이 맞는지 의구심이 들기 시작한다. 그래서 1개월간 식빵을 사운 뒤 매일 무게를 기록하였고 그 결과는 `bread` 데이터셋으로 저장하였다.

25일간 측정한 빵의 평균 무게는 `480g`이었다. 이 값을 가지고 식빵의 무게가 500g이 되지 않는다며 베이커리를 찾아가 컴플레인을 해야 할까?

## #01. 준비작업

### 1. 라이브러리 참조

In [1]:
from jussam import load_data
from helpers import my_stats

📦 연세대학교 주영아 교수가 제작한 라이브러리를 사용중입니다.
📧 Email(1): j.purplerose@yonsei.ac.kr
📧 Email(2): j.purplerose@gmail.com
📝 Website: https://juyounga.kr/


### 2. 데이터 가져오기

In [2]:
origin = load_data('bread')
origin.head()

📚 매일 아침 동네 빵집에서 식빵을 구입해 무게를 기록한 데이터


,무게
0,482
1,479
2,485
3,474
4,479


## #02. 단일표본 T-Test

### 1. 빵의 평균 무게가 500g과 다른지 확인

| 구분               | 수식            | 내용                                       |
| ------------------ | --------------- | ------------------------------------------ |
| 빵의 평균 무게     | $\mu$           |                                            |
| 귀무가설($H_0$)    | $\mu = 500$     | 빵의 평균 무게가 `500`과 차이가 없다.      |
| 대립가설($H_1$)    | $\mu \neq 500$  | 빵의 평균 무게가 `500`과 차이가 있다.      |

In [3]:
my_stats.test_1sample(origin, '무게', 500)

statistic  p-value  significant   result
test              alternative                                          
One-sample t-test two-sided      -37.524    0.000         True  μ ≠ 500
                  less           -37.524    0.000         True  μ < 500
                  greater        -37.524    1.000        False  μ ≤ 500

### 2. 결론

검정 결과는 다음과 같다.

- **양측검정(two-sided)**: 검정통계량 `-37.524`, p-value `0.000` → 유의수준 `0.05`보다 작으므로 귀무가설($\mu = 500$)을 기각한다. 즉, **빵의 평균 무게는 500g과 통계적으로 유의한 차이가 있다.**
- **단측검정(less)**: p-value `0.000` → 빵의 평균 무게가 **500g보다 유의하게 작다.**
- **단측검정(greater)**: p-value `1.000` → 500g보다 크다고 볼 근거는 없다.

**답변:** 25일간 측정한 평균 무게 `480g`은 단순한 우연(측정 오차)으로 보기 어렵다. 단일표본 T검정 결과 p-value가 `0.05`보다 훨씬 작아(`< 0.001`) 빵의 평균 무게가 표시값 `500g`보다 통계적으로 유의하게 **가볍다**는 것이 확인되었다. 따라서 베이커리를 찾아가 컴플레인을 제기할 **충분한 통계적 근거가 있다.**